File: broadcastreceiverlibrary/src/main/java/com/osapps/broadcastreceiverlibrary/utils/IntentFilterProducer.java
package com.osapps.broadcastreceiverlibrary.utils;

import android.content.IntentFilter;

import java.util.Random;

/**
 * Created by osapps on 09/05/2018.
 */
public class IntentFilterProducer {


    public static IntentFilter produceFilter(String filterName) {
        IntentFilter intentFilter = new IntentFilter();
        intentFilter.addAction(filterName);
        intentFilter.setPriority(new Random().nextInt(100));
        return intentFilter;
    }
}

File: broadcastreceiverlibrary/src/main/java/com/osapps/broadcastreceiverlibrary/utils/IntentFilterProducer.kt
package com.osapps.broadcastreceiverlibrary.utils

import android.content.IntentFilter

import java.util.Random

/**
 * Created by osapps on 09/05/2018.
 */
fun produceFilter(filterName: String): IntentFilter {
        val intentFilter = IntentFilter()
        intentFilter.addAction(filterName)
        intentFilter.priority = Random().nextInt(100)
        return intentFilter
    }
